### PR TITLE
feat: authentication screen

### PR DESCRIPTION
**Pull request by DavidZZGi**
## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: lib/app/routes/router/app_router.gr.dart [hunk 1]
```diff
@@ -54,7 +54,7 @@ abstract class _$AppRouter extends RootStackRouter {
       final args = routeData.argsAs<SignInRouteArgs>();
       return AutoRoutePage<dynamic>(
         routeData: routeData,
-        child: SignInPage(
+        child: NewSignInPage(
```
Comment (yarn-rp) [inline lib/app/routes/router/app_router.gr.dart:57]: This is an autogenerated file, it says do not modify by hand on the top file. If you want to change a route, you do it on the `app_router.dart`. If you have any questions about this, please share it on slack. We'll be very happy to answer your questions

File: lib/authentication/presentation/new_sign_in.dart [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                          if (email.isNotEmpty && password.isNotEmpty) {
+                            signInCubit.signInWithEmailAndPassword(
+                              email,
+                              password,
+                            );
+                          }
+                        },
+                      ),
+                      const SizedBox(height: AppSpacing.xlg),
+                      const Text("Don't have an account?"),
```
Comment (yarn-rp) [inline lib/authentication/presentation/new_sign_in.dart:125]: We are going to start using `l10n` for localized text. Although English is the primary language to support, the app currently supports `en` and `es` as languages. We could stick to english on both files if we don't want to take care of the translations now, so you could fill the same texts on both files. If you have some doubts you can read the [documentation](https://docs.flutter.dev/ui/accessibility-and-internationalization/internationalization) or ask me via slack

File: lib/authentication/presentation/new_sign_in.dart [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+              child: Padding(
+                padding: const EdgeInsets.all(16),
+                child: Container(
+                  padding: const EdgeInsets.all(20),
+                  width: MediaQuery.of(context).size.width * 0.4,
+                  child: Column(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    children: [
+                      BlueprintLogo(
+                        width: MediaQuery.of(context).size.width * 0.3,
```
Comment (yarn-rp) [inline lib/authentication/presentation/new_sign_in.dart:77]: This is no longer the optimal way to do this. With latest flutter versions, you can do 
`MediaQuery.sizeOf(context)` which is way better in terms of performance. 

To help with reutilization, usually you should compute all the context dependent variables at the top of the method, that way you don't get a new instance for everything you are going to use

**Review by yarn-rp — CHANGES_REQUESTED** [review]
Please let me know if you need more clarification on these comments. We could always sync via slack if something is not clear

File: lib/authentication/presentation/new_sign_in.dart [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+import 'package:auto_route/auto_route.dart';
+import 'package:blueprint/app/dependency_injection/init.dart';
+import 'package:blueprint/app/routes/guards/authentication_guard.dart';
+import 'package:blueprint/app/routes/routes.dart';
+import 'package:blueprint/authentication/state_management/sign_in_cubit/sign_in_cubit.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+@RoutePage()
+class NewSignInPage extends StatelessWidget {
```
[the remark this answers is not in the document]
Comment (DavidZZGi) [reply lib/authentication/presentation/new_sign_in.dart]: yep, that's i was asking yesterday, i think removing the old ones and change the prefix of the new ones

File: lib/app/routes/router/app_router.gr.dart (hunk 1 above)
Comment (DavidZZGi) [reply lib/app/routes/router/app_router.gr.dart:57]: ok

File: lib/authentication/presentation/new_sign_in.dart (hunk 3 above)
Comment (DavidZZGi) [reply lib/authentication/presentation/new_sign_in.dart]: ok

File: lib/authentication/presentation/pages/sign_in_page.dart [hunk 5]
```diff
@@ -25,75 +27,122 @@ class SignInPage extends StatelessWidget {
...
-  final AuthenticationResultFunction onResult;
+
+  bool _showPassword = true;
 
   @override
   Widget build(BuildContext context) {
     final signInCubit = context.watch<SignInCubit>();
-
+    final size = MediaQuery.sizeOf(context);
+    final lang = AppLocalizations.of(context);
```
Comment (yarn-rp) [inline lib/authentication/presentation/pages/sign_in_page.dart:52]: nit: very dummy change, but usually this variable is named `l10n`.  

pdf: l10n means localization, it's like l(ocalizatio=10 chars)n

File: lib/authentication/presentation/pages/sign_in_page.dart [hunk 6]
```diff
@@ -25,75 +27,122 @@ class SignInPage extends StatelessWidget {
...
+                          if (email.isNotEmpty && password.isNotEmpty) {
+                            signInCubit.signInWithEmailAndPassword(
+                              email,
+                              password,
+                            );
+                          }
+                        },
+                      ),
+                      const SizedBox(height: AppSpacing.xlg),
+                      Text(lang.do_not_have_account),
```
Comment (yarn-rp) [inline lib/authentication/presentation/pages/sign_in_page.dart:127]: the recommended naming convention is PascalCase, not snake case. See the example counter title. Also it's always nice to add the description to help with ambiguity

File: lib/core/l10n/arb/app_es.arb [hunk 7]
```diff
@@ -3,5 +3,14 @@
     "counterAppBarTitle": "Contador",
     "@counterAppBarTitle": {
         "description": "Texto mostrado en la AppBar de la página del contador"
-    }
+    },
+    "do_not_have_account":"No tienes una cuenta?",
```
Comment (yarn-rp) [inline lib/core/l10n/arb/app_es.arb:7]: I would prefer to leave this all in english, that way we don't waste time. Just use the same message we used on the `app_en` counterpart and good to go.

File: lib/core/l10n/arb/app_en.arb [hunk 8]
```diff
@@ -3,5 +3,15 @@
     "counterAppBarTitle": "Counter",
     "@counterAppBarTitle": {
         "description": "Text shown in the AppBar of the Counter Page"
-    }
+    },
+    
+    "do_not_have_account":"Don't have an account?",
+    "forget_pass":"Forget Password",
```
Comment (yarn-rp) [inline lib/core/l10n/arb/app_en.arb:9]: this should be forgot password I think, not forget.

**Review by yarn-rp — CHANGES_REQUESTED** [review]

File: firebase.json
Comment (yarn-rp) [inline firebase.json:1]: I thought we agreed on not including changes on this file. Why are we still seeing this?

File: lib/authentication/presentation/pages/sign_in_page.dart [hunk 9]
```diff
@@ -25,75 +27,122 @@ class SignInPage extends StatelessWidget {
...
+                } else if (state is SignInError) {
+                  ScaffoldMessenger.of(context).showSnackBar(
+                    SnackBar(
+                      content: Text(state.failure.toString()),
+                    ),
+                  );
+                }
+              },
+              child: Padding(
+                padding: const EdgeInsets.all(16),
```
Comment (yarn-rp) [inline lib/authentication/presentation/pages/sign_in_page.dart:71]: Use `AppSpacings` for paddings too

File: lib/authentication/presentation/pages/sign_up_page.dart [hunk 10]
```diff
@@ -24,86 +27,152 @@ class SignUpPage extends StatelessWidget {
...
+                padding: const EdgeInsets.all(16),
+                child: Container(
+                  padding: const EdgeInsets.all(20),
+                  width: size.width * 0.4,
+                  child: Column(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    children: [
+                      BlueprintLogo(
+                        width: size.width * 0.3,
+                        height: size.height * 0.15,
```
Comment (yarn-rp) [inline lib/authentication/presentation/pages/sign_up_page.dart:85]: I'm not a fan of this code. It looks weird on small devices, like too much height. I think if we only provide the width and let the height adjust by the aspect ratio of the image should work better. Let me know if this doesn't work but I think should work better

**Review by yarn-rp — CHANGES_REQUESTED** [review]

**Review by yarn-rp — APPROVED** [review]

**Pull request merged**